src/test/java/org/commonjava/util/partyline/JoinFileWriteAndCloseBeforeFinishedTest.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc. (jdcasey@commonjava.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.util.partyline;
+
+import org.jboss.byteman.contrib.bmunit.BMScript;
+import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CountDownLatch;
+
+@RunWith( org.jboss.byteman.contrib.bmunit.BMUnitRunner.class )
+@BMUnitConfig( loadDirectory = "target/test-classes/bmunit", debug = true )
+public class JoinFileWriteAndCloseBeforeFinishedTest
+        extends AbstractJointedIOTest
+{
+
+    @Test
+    @BMScript( "JoinFileWriteAndCloseBeforeFinished.btm" )
```

Comment:
This is going to be a HUGE pain, I know, but I think I'd prefer to see the rules expressed as annotations directly in the test classes. That way, we can have them located directly with the test they relate to, and we can even add java comments to document what each rule is supposed to do. (Adding those short docs for each rule would be great, too, BTW).

Have a look at this for an example: https://github.com/Commonjava/partyline/pull/47/files#diff-555b698cc3252cc90454b496e054947b